### PR TITLE
feat/config_for_root_people

### PR DESCRIPTION
**Pull request by JarbasAl**
this PR changes some of the mycroft.conf default values to reflect #1 

It also ports https://github.com/MycroftAI/mycroft-core/pull/2872 without changing the order of configs

SYSTEM still has precedence over REMOTE, this was accidentally changed in mycroft-core recently https://github.com/MycroftAI/mycroft-core/issues/3029 but i consider that a bug, if https://github.com/MycroftAI/mycroft-core/pull/2861 gets merged we can revisit this, for now i did not port that change

the dictionary utils used in this PR can be found here https://github.com/OpenVoiceOS/zzz-old-ovos-utils/pull/66

```
  // system administrators can define different constraints in how configurations are loaded
  // this is a mechanism to require root to change these config options
  "system": {
    // do not allow users to tamper with settings at all
    "disable_user_config": false,
    // do not allow remote backend to tamper with settings at all
    "disable_remote_config": false,
    // protected keys are individual settings that can not be changed at remote/user level
    // nested dictionary keys can be defined with "key1:key2" syntax,
    // eg. {"a": {"b": True, "c": False}}
    // to protect "c" you would enter "a:c" in the section below
    "protected_keys": {
        // NOTE: selene backend expects "opt_in" to be changeable in their web ui
        // that effectively gives them a means to enable spying without your input
        // Mycroft AI can be trusted, but you dont need to anymore!
        // The other keys are not currently populated by the remote backend
        // they are defined for protection against bugs and for future proofing
        // (what if facebook buys mycroft tomorrow?)
        "remote": [
            "enclosure",
            "server",
            "system",
            "websocket",
            "gui_websocket",
            "network_tests",
            "listener:wake_word_upload:disable",
            "skills:msm:disabled",
            "skills:upload_skill_manifest",
            "skills:auto_update",
            "skills:priority_skills",
            "skills:blacklisted_skills",
            "opt_in"
        ],
        "user": ["system"]
    }
  }
```

### CODE REVIEW

File: mycroft/configuration/config.py
Comment (ChanceNCounter) [inline mycroft/configuration/config.py:208]: [link to key (below)](https://github.com/OpenVoiceOS/ovos-core/pull/19/files#diff-710a53959602f8d38706b48cad5f274c7da04da59f41079d7b84644c7b3535f6R224-R232) for other reviewers' reference

File: mycroft/configuration/config.py
Comment (ChanceNCounter) [inline mycroft/configuration/config.py:267]: this could be an elif (saves the redundant comparison)

File: mycroft/configuration/config.py
Comment (ChanceNCounter) [inline mycroft/configuration/config.py:277]: it looks like this was under the if on new L268 (ref old L254)

did the logic flow change? it looks the same up to here.

File: mycroft/configuration/config.py
Comment (ChanceNCounter) [inline mycroft/configuration/config.py:299]: this is beautiful

should it be a function? it would pre-implement additional config layers, though i have no suggestions as to what those might be

File: mycroft/configuration/mycroft.conf
Comment (ChanceNCounter) [inline mycroft/configuration/mycroft.conf:382]: 0¿o;

**Review by ChanceNCounter — COMMENTED** [review]
Read, but not run, nor have I linted the JSON.

Visually, the logic checks out, and the plugin names look right.

File: mycroft/configuration/config.py
Comment (JarbasAl) [reply mycroft/configuration/config.py:277]: that one is always loaded last. previously it was inside the `if` and also inside the `else`, just moved it to remove the duplicated logic

File: mycroft/configuration/mycroft.conf
Comment (JarbasAl) [reply mycroft/configuration/mycroft.conf:382]: mk1 specific code, in the process of being moved https://github.com/OpenVoiceOS/ovos_phal_mk1
the test is related to the arduino only

File: mycroft/configuration/config.py
Comment (JarbasAl) [reply mycroft/configuration/config.py:299]: i can move it into a function if that feels cleaner, this method is already quite big anyway

File: mycroft/configuration/config.py
Comment (JarbasAl) [reply mycroft/configuration/config.py:267]: no it can't, these are not exclusive, the logic for user does not depend on remote being loaded or not

File: mycroft/configuration/config.py
Comment (JarbasAl) [reply mycroft/configuration/config.py:299]: actually, moving this into a func is ugly, it needs `get_xdg_config_locations` and `_get_system_constraints` so those would need to be called twice or passed in args, regardless it would no longer be beautiful :P

File: mycroft/configuration/config.py
Comment (ChanceNCounter) [reply mycroft/configuration/config.py:267]: mea culpa, i misread the indentation level of the else below and it still made sense =P

File: mycroft/configuration/config.py
Comment (ChanceNCounter) [reply mycroft/configuration/config.py:299]: It would certainly need some params, but I don't see those

The interesting part would be creating a clean reference to `skip_n` and `protected_n` for like values of `n`

**Review by ChanceNCounter — APPROVED** [review]

**Comment by ChanceNCounter** [comment]
It wouldn't hurt to wait for another review, but my questions are answered

**Comment by AIIX** [comment]
All looking good to me except "the dictionary utils used in this PR can be found here OpenVoiceOS/zzz-old-ovos-utils#66" does this mean we need to depend on that archived util or is that ported over to this or to newer ovos utils?

**Comment by JarbasAl** [comment]
> All looking good to me except "the dictionary utils used in this PR can be found here [OpenVoiceOS/zzz-old-ovos-utils#66](https://github.com/OpenVoiceOS/zzz-old-ovos-utils/pull/66)" does this mean we need to depend on that archived util or is that ported over to this or to newer ovos utils?

i was a github accident..... the PR was made in the old repo, but there is still only 1 ovos_utils package, the other repo should be ignored

**Pull request merged**
